src/screens/dialogs/move-file-dialog.tsx

```diff
@@ -0,0 +1,154 @@
+import { listFilesService } from "../../services/files/list-file.service";
+import { useSearchParams } from "react-router-dom";
+import { Dialog } from "../../components/Dialog";
+import { File } from "../../types/entities";
+import { FolderPlus } from "lucide-react";
+import {
+  FilesDialogsContext,
+  AVAILABLE_DIALOGS,
+  AuthContext,
+  FilesContext
+} from "../../context/index";
+import toast from "react-hot-toast";
+
+export const MoveFileDialog = () => {
+  const { removeFile } = useContext(FilesContext);
+  const [searchParams, _setSearchParams] = useSearchParams();
+  const { dialogsVisibilityState, closeDialog, selectedFile } =
+    useContext(FilesDialogsContext);
+  const Open = dialogsVisibilityState[AVAILABLE_DIALOGS.MOVE_FILE];
+  const { session } = useContext(AuthContext);
+  const [folders, setFolders] = useState<File[]>([]);
+  const [moving, setMoving] = useState(false);
+  const [selectedFolder, setSelectedFolder] = useState<File | null>(null);
+  const directory = searchParams.get("directory");
+
+  useEffect(() => {
+    if (Open && selectedFile) {
+      const fetchFolders = async () => {
+        const { success, ...res } = await listFilesService({
+          token: session?.token as string,
+          directory: selectedFolder ? selectedFolder.uuid : directory
+        });
+
+        if (!success) {
+          toast.error(res.msg);
+        } else {
+          const folderList = res.files.filter(
+            (file) => !file.isFile && file.uuid !== selectedFile?.uuid
+          );
+          setFolders(folderList);
+        }
+      };
+
+      fetchFolders();
+    }
+  }, [directory]);
+
+  const handleMove = async () => {
+    if (moving || !selectedFolder || !selectedFile) return;
+
+    if (!selectedFolder.isFile) {
+      setMoving(true);
+      const { success, ...res } = await moveFileService({
+        directoryUUID: selectedFolder.uuid,
+        fileUUID: selectedFile.uuid as string,
+        token: session?.token as string
+      });
+
+      if (!success) {
+        toast.error(res.msg);
+        setMoving(false);
+        return;
+      }
+
+      const successMessage = selectedFile.isFile
+        ? "File has been moved successfully"
+        : "Folder has been moved successfully";
+
+      toast.success(successMessage);
+      setMoving(false);
+      closeDialog(AVAILABLE_DIALOGS.MOVE_FILE);
+      setSelectedFolder(null);
+      removeFile(selectedFile.uuid);
+    } else {
+      toast.error("This action cannot be performed");
+    }
+  };
+
+  const changeDirectory = (directoryUUID: string) => {
+    if (!selectedFile) return;
+
+    const selectedFolder = folders.find(
+      (folder) => folder.uuid === directoryUUID
+    );
+    if (!selectedFolder) {
+      return;
+    }
+
+    setSelectedFolder(selectedFolder);
+    setFolders([]);
+
+    loadFolderContents(directoryUUID);
+  };
+
+  const loadFolderContents = async (directoryUUID: string) => {
```

Comment:
This is the same as the `fetchFolders` function in the `useEffect`. Adding the `selectedFolder` to the dependencies array should make this function unnecessary.